lib/order.rb

```diff
@@ -0,0 +1,59 @@
+require 'csv'
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+  
+  FULFILLMENT_VALID_STATUS = [:paid, :processing, :shipped, :complete, :pending]
+  
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    
+    raise ArgumentError if !FULFILLMENT_VALID_STATUS.include?(fulfillment_status)
+  end
+  
+  def total
+    sum = products.values.sum
+    tax = sum * 0.075
+    grand_total = sum + tax
+    return grand_total.round(2)
+  end
+  
+  def add_product(name, price)
+    raise ArgumentError if products.key?(name)
+    @products[name] = price
+  end
+  
+  def self.products_hash(orders)
+    products = {}
+    items = orders.split(";")
+    items.each do |item|
+      product_info = item.split(":")
+      products[product_info[0]] = product_info[1].to_f
+    end
+    return products
+  end
+  
+  def self.all
+    order_info = CSV.read('../data/orders.csv')
```

Comment:
Judging by this, your ran your `rake` from inside the `lib` folder. It's not a penalty, but in the future I'd suggest running rake from the root of the project, as that is what the instructors do. :)